sockets.tcp.js

```diff
@@ -95,7 +113,13 @@ exports.secure = function(socketId, options, callback) {
         callback(0);
     };
     var fail = callback && function(error) {
-        exports.onReceiveError.fire(error);
+            var sendInfo = {
+                bytesSent: 0,
+                resultCode: error.resultCode,
+                message: error.message,
+                socketId: socketId
+            };
+            exports.onReceiveError.fire(sendInfo);
```

Comment:
Can we just create a function to do this? They are all the same so from a maintainability POV - if we need to change the object structure, we'd need to do the change 4 times. If we had a function it would be just the once.

Reply:
Can do